qualtricssurvey/qualtricssurvey.py

```diff
@@ -13,49 +13,105 @@
-        Gather scenarios to be displayed in the workbench
-        """
-        return [
-            ('Qualtrics Survey',
-             """<sequence_demo>
-                    <qualtricssurvey />
-                    <qualtricssurvey name="My First XBlock" />
-                </sequence_demo>
-             """),
-        ]
-
-    name = String(
-        default='Qualtrics Survey',
+    display_name = String(
+        default="Qualtrics Survey",
+        scope=Scope.settings,
+    )
+    survey_id = String(
+        default="Enter your survey ID here.",
+        scope=Scope.settings,
+    )
+    your_university = String(
+        default="stanford",
+        scope=Scope.settings,
+    )                       
+    link_text = String(
+        default="click here",
+        scope=Scope.settings,
+    )
+    param_name = String(
+        default="a",
         scope=Scope.settings,
-        help="This is the XBlock's name",
     )
 
     def student_view(self, context=None):
         """
         Build the fragment for the default student view
         """
+
+        display_name = self.display_name
+        survey_id = self.survey_id
+        your_university = self.your_university
+        link_text = self.link_text
+        param_name = self.param_name
+
+        anon_user_id = str(self.xmodule_runtime.anonymous_student_id)
+
+        # %%PARAM%% substitution only works in HTML components
+        # so it has to be done here for USER_ID
+        user_id_string = ""
+        if param_name:
+            user_id_string = ('&amp;{param_name}={anon_user_id}').format(
+                param_name=param_name,
+                anon_user_id=anon_user_id,
+            )
+
+        html_source = self.get_resource_string('view.html')
+        html_source = html_source.format(
+            self=self,
+            display_name=display_name,
+            survey_id=survey_id,
+            your_university=your_university,
+            link_text=link_text,
+            user_id_string=user_id_string,
+        )
+
         fragment = self.build_fragment(
-            path_html='view.html',
-            path_css='view.less.min.css',
-            path_js='view.js.min.js',
-            fragment_js='QualtricsSurveyView',
+            html_source=html_source,
         )
+
         return fragment
 
     def studio_view(self, context=None):
         """
         Build the fragment for the edit/studio view
+        """
 
-        Implementation is optional.
+        display_name = self.display_name
+        survey_id = self.survey_id
+        your_university = self.your_university
+        link_text = self.link_text
+        param_name = self.param_name
+
+        # Show the %%USER_ID%% string in the source view
+        user_id_string = ""
+        if param_name:
+            user_id_string = ('&amp;{param_name}={anon_user_id}').format(
+                param_name=param_name,
+                anon_user_id='%%USER_ID%%',
+            )
+
+        source_text = """
```

Comment:
I think this would look a little better as a single triple-quoted string with parameters that was then formatted.  Otherwise, it would be nice to avoid the single line triple-quoted strings like `"""qualtrics.com/SE/?SID="""`.

```
source_text = """
    <p>The survey will open in a new browser tab or window.</p>
    <p>
        <a href="https://{university}.qualtrics.com/SE/?SID={survey_id}{user_id} target="_blank">{link_text}</a>
    </p>
"""
source_text = source_text.format(university=your_university, ...)
```